lineapy/graph_reader/artifact_collection.py

```diff
@@ -75,24 +77,37 @@ def __init__(
                 )
 
             # Retrieve artifact
+            if args["artifact_name"] in self.art_name_to_node_id.keys():
+                logger.error("%s is duplicated", args["artifact_name"])
+                raise KeyError("%s is duplicated", args["artifact_name"])
             try:
                 version = args.get("version", None)
                 if version is None:
-                    art = get(artifact_name=args["artifact_name"])
+                    art_orm = self.db.get_artifact_by_name(
+                        artifact_name=args["artifact_name"]
+                    )
                 else:
-                    art = get(
+                    art_orm = self.db.get_artifact_by_name(
                         artifact_name=args["artifact_name"],
                         version=int(version),
                     )
-                if args["artifact_name"] in self.art_name_to_node_id.keys():
-                    logger.error("%s is duplicated", args["artifact_name"])
-                    raise KeyError("%s is duplicated", args["artifact_name"])
-                self.art_name_to_node_id[args["artifact_name"]] = art._node_id
-                self.node_id_to_session_id[art._node_id] = art._session_id
             except Exception as e:
                 logger.error("Cannot retrive artifact %s", art_entry)
                 raise Exception(e)
 
+            # Construct LineaArtifact object and store its info
+            art = LineaArtifact(
+                db=self.db,
+                _execution_id=art_orm.execution_id,
+                _node_id=art_orm.node_id,
+                _session_id=art_orm.node.session_id,
+                _version=art_orm.version,  # type: ignore
+                name=args["artifact_name"],
+                date_created=art_orm.date_created,  # type: ignore
+            )
+            self.art_name_to_node_id[args["artifact_name"]] = art._node_id
+            self.node_id_to_session_id[art._node_id] = art._session_id
+
```

Comment:
FYI: We actually do not need the entire `LIneaArtifact` and its attributes, but I decided to still use it here as it is a common interface used in the codebase to pass around artifacts. Hence, we prefer consistency.

Reply:
I personally think it makes more sense to remane `db.get_artifact_by_name -> ArtifactORM` to `db.get_artifact_orm -> ArtifactORM` and create a new `db.get_artifact_by_name -> LineaArtifact` that wrap this code block. However, I can live with this for now.

Reply:
@mingjerli Yes, on the same page with you. Given that this type of code (i.e., getting artifact ORM with existing `db.get_artifact_by_name()` and wrapping it into `LineaArtifact`) is present in other parts of the codebase (e.g., inside `api.get()`), I would rather tackle this change/refactor in a separate PR. Are we good with this?

Reply:
LineaArtifact is the user facing object. for all internal stuff we pass around artifactORM. so i disagree that its a common interface used in the codebase. There is no loss of information when using artifactORM.

Reply:
@lionsardesai I see, yeah, that makes sense. I will then update the code here to use `ArtifactORM`.

@mingjerli I will have to update relevant code in `session_artifacts.py`. Let me know if you have concerns about me making this update in this PR.